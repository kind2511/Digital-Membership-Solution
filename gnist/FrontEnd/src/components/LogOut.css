.logout-modal {
  position: fixed;
  z-index: 1000;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0,0,0,0.4);
}

.logout-modal-content {
  background-color: yellow;
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 4px 8px rgba(0,0,0,0.2);
  text-align: center;
  width: auto;
  min-width: 300px; /* Minimum width */
  max-width: 400px; /* Maximum width */
  margin: auto;
}

.logout-modal-logo {
  margin-bottom: 20px;
  display: block;
  max-width: 100px;
  margin-left: auto;
  margin-right: auto;
}

.logout-modal-buttons {
  position: relative;
}

.logout-button {
  background-color: black;
  color: yellow;
  border: none;
  padding: 10px 20px;
  margin: 10px; /* Space between the buttons */
  cursor: pointer;
  font-weight: bold;
  border-radius: 5px;
}

.logout-button:hover {
  opacity: 0.8;
}
