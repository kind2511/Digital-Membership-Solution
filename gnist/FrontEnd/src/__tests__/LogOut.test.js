import React from 'react';
import renderer from 'react-test-renderer';
import { Auth0Context } from '@auth0/auth0-react';
import LogOut from '../components/LogOut';

// Mock Auth0 context
const mockAuth0Context = {
  logout: jest.fn(),
};

test('LogOut component snapshot', () => {
  const component = renderer.create(
    <Auth0Context.Provider value={mockAuth0Context}>
      <LogOut />
    </Auth0Context.Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
