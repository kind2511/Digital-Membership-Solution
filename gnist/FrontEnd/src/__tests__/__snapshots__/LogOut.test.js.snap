// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogOut component snapshot 1`] = `
<button
  className="logout-button"
  onClick={[Function]}
>
  Log Out
</button>
`;
